src/transformStateWithClones.js

```diff
@@ -8,6 +8,24 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateHistory = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateHistory, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete stateHistory[key];
+      }
+    } else if (action.type === 'clear') {
+      stateHistory = {};
+    }
```

Comment:
The task suggests using a `switch` statement with a default case for better readability and error handling, instead of `if-else` statements. Consider refactoring this part of the code to use a `switch` statement.